modules/mesh-task/iam.tf

```diff
@@ -0,0 +1,130 @@
+locals {
+  // We use a splat trick to workaround a Terraform limitation: `The "count" value
+  // depends on resource attributes that cannot be determined until apply`
+  //
+  // Basically, users will pass in the output value of another resource. That resource
+  // may not be created during the planning phase, so Terraform cannot inspect the value
+  // to set a `count` field. So it errors.
+  //
+  // The workaround uses a splat to convert to 1-length or 0-length list.
+  //
+  // "If the value is anything other than a null value then the splat expression will transform
+  // it into a single-element list...If the value is null then the splat expression will return
+  // an empty tuple."
+  // https://www.terraform.io/docs/language/expressions/splat.html#single-values-as-lists
+  create_task_role      = length(var.task_role[*]) == 0
+  create_execution_role = length(var.execution_role[*]) == 0
+}
```

Comment:
Could you also add

```hcl
  execution_role_id = local.create_execution_role ? aws_iam_role.execution[0].id : var.execution_role.id
  task_role_id = local.create_task_role ? aws_iam_role.task[0].id : var.task_role.id
```

So you don't duplicate this ternary in other places? Or does that not work.